### PR TITLE
Adjust ``entropy_tol`` to avoid over-refinement

### PR DESCRIPTION
**Pull request by maximilianmetti**
updated refinement to adjust tolerance dynamically to stay below maximal mesh size

### CODE REVIEW

**Comment by maximilianmetti** [comment]
ping @arthbous

**Comment by arthbous** [comment]
The problem with that, it refines the first cells it finds and no cells after the max_cells has been attained. In this case it could refines cells that are less important than other, we should implement the priority or hierarchy.

File: src/domains.cpp [hunk 1]
```diff
@@ -333,15 +308,24 @@ unsigned int check_electric_field (dolfin::Function *voltage,
...
-        marked_elem_count++;
-        cell_marker.values()[errVecInd] = true;
+    if (error_vector[errVecInd] > new_entropy_tol) {
+      marked_elem_count++;
+      cell_marker.values()[errVecInd] = true;
+      
+      if (marked_elem_count > max_elements) {
+        cell_marker.values()[errVecInd] = false;
+        new_entropy_tol *= 5.0;
+        errVecInd = 0;
```
Comment (maximilianmetti) [inline src/domains.cpp]: this should restart the ``for`` loop.  This algorithm should increase the ``new_entropy_tol`` and then restart the algorithm...
I see that I'm missing
```
marked_elem_count = 0;
```

**Comment by maximilianmetti** [comment]
It looks like it's working properly now

**Comment by maximilianmetti** [comment]
Also, since the ``new_entropy_tol`` is reset and the loop is re-run, it only refines the worst elements

File: src/domains.cpp (hunk 1 above)
Comment (arthbous) [inline src/domains.cpp]: Wouldn't be better to put ''break;'' ?

Comment (maximilianmetti) [inline src/domains.cpp]: that wouldn't put us back to the top of the for loop though... This makes the for-loop start over.

File: src/domains.cpp [hunk 2]
```diff
@@ -191,15 +181,25 @@ unsigned int check_local_entropy (dolfin::Function *cation,
...
+  for (int errVecInd = 0; errVecInd < error_vector.size(); errVecInd++) {
+    if (error_vector[errVecInd] > new_entropy_tol) {
+      marked_elem_count++;
+      cell_marker.values()[errVecInd] = true;
+      
+      if (marked_elem_count > max_elements) {
+        cell_marker.values()[errVecInd] = false;
+        new_entropy_tol *= 5.0;
+        errVecInd = -1;
+        marked_elem_count = 0;
```
Comment (arthbous) [inline src/domains.cpp]: I would add  ''cell_marker.set_all(false)''.

File: src/domains.cpp [hunk 3]
```diff
@@ -333,15 +309,24 @@ unsigned int check_electric_field (dolfin::Function *voltage,
...
-        marked_elem_count++;
-        cell_marker.values()[errVecInd] = true;
+    if (error_vector[errVecInd] > new_entropy_tol) {
+      marked_elem_count++;
+      cell_marker.values()[errVecInd] = true;
+      
+      if (marked_elem_count > max_elements) {
+        cell_marker.values()[errVecInd] = false;
+        new_entropy_tol *= 5.0;
+        errVecInd = -1;
```
Comment (arthbous) [inline src/domains.cpp]: I would add  ``cell_marker.set_all(false)`` and ``marked_elem_count=0``

**Comment by maximilianmetti** [comment]
That should do it!

**Pull request merged**